nuxeo-runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/model/DefaultComponent.java

```diff
@@ -33,13 +34,20 @@
      */
     protected Long lastModified;
 
+    /**
+     * @since 10.3
+     */
+    protected DescriptorRegistry registry = new DescriptorRegistry();
```

Comment:
But that means there's a full registry per component. I thought we would have a single registry somewhere global in the framework so that we can enumerate everything from there if needed.

Reply:
TBD

Reply:
As discussed, the registry instantiation will be moved out of DefaultComponent and accessed dynamically.